File: src/pydjinni/generator/java/it/java/interface/tests/TestOptionalInterface.java
// Copyright 2025 jothepro
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;
import test.interface_test.OptionalInterface;

class TestOptionalInterface {
    OptionalInterface instance;

    @BeforeEach
    void setup() {
        instance = OptionalInterface.getInstance();
        assertNotNull(instance);
    }

    @Test
    void testNullInterface() {
        var nullInstance = OptionalInterface.getNullInstance();
        assertNull(nullInstance);
    }

    @Test
    void testOptionalParameter() {
        var result = instance.optionalParameter("some optional string");
        assertNotNull(result);
        assertEquals("some optional string", result);
    }

    @Test
    void testOptionalNullParameter() {
        var result = instance.optionalNullParameter(null);
        assertNull(result);
    }



}

File: src/pydjinni/generator/java/it/kotlin/interface/tests/TestOptionalInterface.kt
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import test.interface_test.OptionalInterface

class TestOptionalInterface {
    private var instance: OptionalInterface? = OptionalInterface.getInstance()

    @Test
    fun testNullInterface() {
        val nullInstance: OptionalInterface? = OptionalInterface.getNullInstance()
        assertNull(nullInstance)
    }

    @Test
    fun testOptionalParameter() {
        val result: String? = instance?.optionalParameter("some optional string")
        assertNotNull(result)
        assertEquals("some optional string", result)
    }

    @Test
    fun testOptionalNullParameter() {
        val result: String? = instance?.optionalNullParameter(null)
        assertNull(result)
    }
}
